src/views/server_setup_views.py

```
@@ -0,0 +1,142 @@
+"""
+Server Setup View.
+
+Contains all the view classes for views used in the /server-setup command.
+"""
+import logging
+import discord
+
+from discord.ui import View
+
+from src.logic import database_connection, slash_logic
+
+from os import environ
+logger = logging.getLogger(environ['LOGGER_NAME'])
+
+class ServerCreationView(View):
+    def __init__(self):
+        super().__init__()
+
+    @discord.ui.button(label = "Add", style = discord.ButtonStyle.green, row = 2)
+    async def add_guild(self, button, interaction):
+        try:
+            await database_connection.add_guild(interaction.guild_id, interaction.guild.name)
+            success = True
+        except Exception as e:
+            logger.error(e)
+            success = False
+
+        if success:
+            await interaction.response.send_message("Guild Added to the DB")
+            button.disable()
+        else:
+            await interaction.response.send_message("There was an error adding the guild to the DB. Please try again, or contact a Bot Developer for assistance")
+    
+    @discord.ui.button(label = "Remove", style = discord.ButtonStyle.red, row = 2)
+    async def remove_guild(self, button, interaction: discord.Interaction):
+        try:
+            await database_connection.remove_guild(interaction.guild_id)
+            success = True
+        except Exception as e:
+            logger.error(e)
```

You should never just "catch all exceptions". Especially when you aren't actually handling anything, just printing the error to the log. Catch the exception that you excpect is possible to throw.